toolkit/tools/scheduler/schedulerutils/buildworker.go

```diff
@@ -84,7 +85,7 @@ func BuildNodeWorker(channels *BuildChannels, agent buildagents.BuildAgent, buil
 }
 
 // buildBuildNode builds a TypeBuild node, either used a cached copy if possible or building the corresponding SRPM.
-func buildBuildNode(node *pkggraph.PkgNode, pkgGraph *pkggraph.PkgGraph, agent buildagents.BuildAgent, canUseCache bool, buildAttempts int) (usedCache bool, builtFiles []string, logFile string, err error) {
+func buildBuildNode(node *pkggraph.PkgNode, pkgGraph *pkggraph.PkgGraph, graphMutex sync.RWMutex, agent buildagents.BuildAgent, canUseCache bool, buildAttempts int) (usedCache bool, builtFiles []string, logFile string, err error) {
```

Comment:
Is it possible that `node *pkggraph.PkgNode` will be referring to a node which has been modified? ie does the mutex need to be taken farther out?

Reply:
I'm assuming that once a node is popped of the graph the data in it won't be modified. ie, if we could start a build before, we should still be able to finish that build.

Reply:
Once a node has been given to a worker, ownership of the node itself is transferred to that worker and that worker only. Other routines may alter the edges connecting to that node, but not the node itself.